docs/sphinx/contribution/code-style-python.rst

```diff
@@ -0,0 +1,199 @@
+Python Coding Style Guide
+=========================
+
+CHIPSEC mostly follows the PEP8 with some exceptions.  This attempts to highlight those as well as clarify others.
+
+Consistency and readability are the goal but not at the expense of readability or functionality.
+
+If in doubt, follow the existing code style and formatting.
+
+
+#. PEP8
+
+    PEP 8 is a set of recommended code style guidelines (conventions) for Python.
+
+    `PEP 8 <https://www.python.org/dev/peps/pep-0008/>`_
+
+
+#. Linting tools
+
+    CHIPSEC includes a Flake8 configuration file
+
+    `CHIPSEC flake8 config <https://github.com/chipsec/chipsec/blob/master/.flake8>`_
+
+
+#. Zen of Python
+
+    Great philosophy around Python building principles.
+
+    `PEP 20 <https://www.python.org/dev/peps/pep-0020/>`_
+
+
+#. Headers and Comments
+
+    Use single line comments, a single hash/number sign/octothorpe '#'.
+
+    Should contain a space immediately after the '#'.
+
+    .. code-block:: python
+
+        # Good header comment
+
+
+#. Single vs Double Quotes
+
+    Single quotes are encouraged but can vary with use case.
+
+    Avoid using backslashes '\\' in strings.
+
+    .. code-block:: python
+
+        'This is a preferred "string".'
+        "Also an acceptable 'string'."
+
+        "Avoid making this \"string\"."
+
+
+#. Imports
+
+    Import order:
+        #. Python standard library
+        #. Third-party imports
+        #. CHIPSEC and local application imports
+
+    Avoid using ``import *`` or ``from import *``.  This could pollute the namespace.
+
+    .. code-block:: python
+
+        # Good
+        import sys
+        from chipsec.module_common import BaseModule, ModuleResult
+
+        # Bad - using '*' and importing sys after local imports
+        import *
+        from chipsec.module_common import *
+        import sys
+
+    Avoid using ``from __future__ imports``.  These may not work on older or all interpreter versions required in all supported environments.
+
+
+#. Line Length
+
+    Maximum line length should be 120 characters.
+
+    If at or near this limit, consider rewriting (eg simplifying) the line instead of breaking it into multiple lines.
+
+    Long lines can be an indication that too many things are happening at once and/or to difficult to read.
```

Comment:
```suggestion
    Long lines can be an indication that too many things are happening at once and/or too difficult to read.
```
After writing the suggestion, I'm double guessing myself. Not sure which is right.